vector/src/main/java/im/vector/app/features/spaces/SpaceCreationActivity.kt

```diff
@@ -51,18 +53,38 @@ class SpaceCreationActivity : SimpleFragmentActivity(), CreateSpaceViewModel.Fac
 
     val viewModel: CreateSpaceViewModel by viewModel()
 
+    private val fragmentLifecycleCallbacks = object : FragmentManager.FragmentLifecycleCallbacks() {
+        override fun onFragmentAttached(fm: FragmentManager, f: Fragment, context: Context) {
+            when (f) {
+                is BetaWarningBottomSheet -> {
+                    f.interactionListener = this@SpaceCreationActivity
+                }
+            }
+            super.onFragmentAttached(fm, f, context)
+        }
+
+        override fun onFragmentDetached(fm: FragmentManager, f: Fragment) {
+            when (f) {
+                is BetaWarningBottomSheet -> {
+                    f.interactionListener = null
+                }
+            }
+            super.onFragmentDetached(fm, f)
+        }
+    }
```

Comment:
We should find a way to improve this code, or at least do not duplicate it everywhere...

Reply:
I changed it to use only inter-fragment communication without going through the hosting activity. 
Looks cleaner as only the two communicating fragments are modified
It's using #setFragmentResultListener and #setFragmentResult